ssc/cmod_hybrid.cpp

```diff
@@ -0,0 +1,762 @@
+
+                    for (int i = 0; i < analysisPeriod && i < (int)count; i++) {
+                        // the cash flow sheets are 1 indexed, batt_rep and replacement_percent is zero indexed
+                        pOMBattReplacementCost[i + 1] = batt_rep[i] * replacement_percent[i] * 0.01 * pOMBattReplacementCost[ i + 1];
+                    }
+                }
+                else
+                {
+                    batt_cap = ((var_table*)compute_module_outputs)->as_double("batt_computed_bank_capacity");
+                    // updated 10/17/15 per 10/14/15 meeting
+                    ssc_number_t* pOMBattReplacementCost = ((var_table*)compute_module_outputs)->allocate("cf_battery_replacement_cost_schedule", analysisPeriod + 1);
+                    escal_or_annual(input, pOMBattReplacementCost, analysisPeriod, "om_batt_replacement_cost", inflation_rate, batt_cap, false, input.as_double("om_replacement_cost_escal") * 0.01);
+                }
+                escal_or_annual(input, pOMCapacity, analysisPeriod, "om_batt_capacity_cost", inflation_rate, batt_cap, false, input.as_double("om_capacity_escal") * 0.01);
+
+                // production O and M conversion to $
+                for (size_t i = 1; i <= (size_t)analysisPeriod; i++)
+                    pOMProduction[i] *= battery_discharged[i - 1];
+
+
+
+                // resize annual outputs
+                size_t arr_length = analysisPeriod + 1;
+                ssc_number_t yr_0_value = 0.0;
+                prepend_to_output((var_table*)compute_module_outputs, "batt_bank_replacement", arr_length, yr_0_value);
+                prepend_to_output((var_table*)compute_module_outputs, "batt_annual_charge_energy", arr_length, yr_0_value);
+                prepend_to_output((var_table*)compute_module_outputs, "batt_annual_discharge_energy", arr_length, yr_0_value);
+                prepend_to_output((var_table*)compute_module_outputs, "batt_annual_charge_from_system", arr_length, yr_0_value);
+
+
+                // add calculations to compute module outputs - done above for regular compute module outputs
+
+                ssc_data_set_table(outputs, compute_module.c_str(), compute_module_outputs);
+                ssc_module_free(module);
+                ssc_data_free(compute_module_outputs);
+
+            }
+
+            bool use_batt_output = false;
+            ssc_number_t* pBattGen = 0;
+            size_t battGenLen = 0;
+            // update gen to battery output
+            if (batteries.size() > 0) {
+                use_batt_output = true;
+                pBattGen = ((var_table*)outputs)->lookup(batteries[0])->table.as_array("gen", &battGenLen);
+            }
+
+ 
+/*
+    //Run grid compute module with fuel cell output
+    run grid
+
+    //Run utilityrate if applicable
+    run utilityrate
```

Comment:
Electricity purchases are not showing up in cash flow for PVWatts-Wind-Battery Single Owner, and maybe others. Unclear if this is the problem or if it is somewhere else. Host-developer looks ok.

SAM file to reproduce: 
[hybrids-electricity-purchases.zip](https://github.com/NREL/ssc/files/12614224/hybrids-electricity-purchases.zip)

Reply:
> Electricity purchases are not showing up in cash flow for PVWatts-Wind-Battery Single Owner, and maybe others. Unclear if this is the problem or if it is somewhere else. Host-developer looks ok.
> 
> SAM file to reproduce: [hybrids-electricity-purchases.zip](https://github.com/NREL/ssc/files/12614224/hybrids-electricity-purchases.zip)

Good catch - issue was that the metering option was set to net metering and would not bill for default configurations - metering option updated to buy all sell all like pv battery configuration and working per attached document.
[GitHubPullRequestsssc_1051.docx](https://github.com/NREL/ssc/files/12618853/GitHubPullRequestsssc_1051.docx)